pkg/controller/v1alpha2/applicationconfiguration/render_test.go

```diff
@@ -920,13 +921,42 @@ func TestSetWorkloadInstanceName(t *testing.T) {
 			reason: "workloadName should not change if already set",
 		},
 		"revisionEnabled true, set revisionName": {
+			traitDefs: []v1alpha2.TraitDefinition{
+				{
+					Spec: v1alpha2.TraitDefinitionSpec{RevisionEnabled: true},
+				},
+			},
+			u: &unstructured.Unstructured{Object: map[string]interface{}{}},
+			c: &v1alpha2.Component{ObjectMeta: metav1.ObjectMeta{Name: "comp"}, Status: v1alpha2.ComponentStatus{LatestRevision: &v1alpha2.Revision{Name: "rev-2"}}},
+			currentWorkload: &unstructured.Unstructured{Object: map[string]interface{}{
+				"metadata": map[string]interface{}{
+					"labels": map[string]string{
+						"app.oam.dev/revision": "rev-v1",
+					},
+				},
+			}},
+			exp: &unstructured.Unstructured{Object: map[string]interface{}{
+				"metadata": map[string]interface{}{
+					"name": "rev-2",
+				},
+			}},
+			reason: "workloadName should align with revisionName",
+		},
+		"revisionEnabled true, don't set revisionName": {
 			traitDefs: []v1alpha2.TraitDefinition{
 				{
 					Spec: v1alpha2.TraitDefinitionSpec{RevisionEnabled: true},
 				},
 			},
 			u: &unstructured.Unstructured{Object: map[string]interface{}{}},
 			c: &v1alpha2.Component{ObjectMeta: metav1.ObjectMeta{Name: "comp"}, Status: v1alpha2.ComponentStatus{LatestRevision: &v1alpha2.Revision{Name: "rev-1"}}},
```

Comment:
```suggestion
			c: &v1alpha2.Component{ObjectMeta: metav1.ObjectMeta{Name: "comp"}, Status: v1alpha2.ComponentStatus{LatestRevision: &v1alpha2.Revision{Name: "rev-2"}}},
```

If I understand well, the test is wrong, what are you trying to test here?

Reply:
If the workload is recreated, the name will be set to `rev-1`, I try to make sure it won't.